aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -342,6 +351,73 @@ module aptos_framework::account {
         rotation_capability
     }
 
+    /// Update the existing capability offer to the `recipient_address` or add a new capability offer if there's none
+    public fun update_or_add_capability_offer<T>(account_address: address, recipient_address: address) acquires Account {
+        let account_resource = borrow_global_mut<Account>(account_address);
+
+        if (type_info::type_of<T>() == type_info::type_of<SignerCapability>()) {
+            if (option::is_some(&account_resource.signer_capability_offer.for)) {
+                option::swap(&mut account_resource.signer_capability_offer.for, recipient_address);
+            } else {
+                option::fill(&mut account_resource.signer_capability_offer.for, recipient_address);
+            };
+        } else if (type_info::type_of<T>() == type_info::type_of<RotationCapability>()) {
+            if (option::is_some(&account_resource.rotation_capability_offer.for)) {
+                option::swap(&mut account_resource.rotation_capability_offer.for, recipient_address);
+            } else {
+                option::fill(&mut account_resource.rotation_capability_offer.for, recipient_address);
+            };
+        } else {
+            abort error::invalid_argument(EINVALID_CAPABILITY_TYPE)
+        };
+    }
```

Comment:
This is actual more expensive and not saving much in terms of reusability. It would be better to inline this afaiu.

Reply:
ooh i see - just updated!